electronics_lib/JLC_Resistor.py

```diff
@@ -0,0 +1,87 @@
+class JLCPCB_ResistorTable(ESeriesResistor):
+  RESISTANCE = PartsTableColumn(Range)  
+  POWER_DISSIPATION = PartsTableColumn(Range)  
+  PRICE = PartsTableColumn(float)
+  FOOTPRINT = PartsTableColumn(str)
+    
+  @classmethod
+  def _generate_table(cls) -> PartsTable:
+    RESISTOR_MATCHES = {
+    'resistance': "\s?(\d+(\.\d*)?[GMkmunp]?)?[\u03A9]\s",
+    'tolerance': "\s?([\u00B1]\d+(\.\d*)?%)\s",
+    'power_dissipation': "\s?([\u00B1]\d+(\.\d*)?ppm)/℃\s"
+    }
+
+    def parse_row(row: PartsTableRow) -> Optional[Dict[PartsTableColumn, Any]]:
+      if (row['Library Type'] != 'Basic'):
+          return None            
+      if (row['First Category'] != 'Resistors'):
+          return None
+
+      new_rows: Dict[PartsTableColumn, Any] = {}
+      try:
+      # handle the footprint first since this is the most likely to filter
+      footprint = ""
+      new_rows[cls.FOOTPRINT] = footprint
+    
+      extracted_values = parse(row['Description'], RESISTOR_MATCHES)
```

Comment:
I think I'd rather prefer the dict be dumped here instead of being declared separately. It kind of would be akin to a magic number, but I don't think the variable name helps us understand what this is doing more than a raw dict, and it also isn't being used anywhere else else.